R/model_diagnostics.R

```diff
@@ -57,5 +82,37 @@ get_model_diagnostic_flags <- function(stan_fit_object,
     flag_high_rhat,
     flag_low_embfi
   )
+
+  # Message if a flag doesn't pass. Still return
+  # the same data, but we want user to know the issue
+  if (any(flag_df[1, ])) {
+    warning("Model flagged for convergence issues, run model diagnostics
+      on the output stanfit object for more information")
+  }
   return(flag_df)
 }
+
+
+#' Method for printing the CmdStan diagnostic summary
+#' for a wwinference_fit object
+#'
+#' @param ww_fit An object of class wwinference_fit
+#' @param ... additional arguments
+#'
+#' @family diagnostics
+#' @export
+diagnostic_summary <- function(ww_fit, ...) {
+  ww_fit$fit$result$diagnostic_summary()
+}
+
+#' Method for printing the CmdStan parameter diagnostics for a
+#' wwinference_fit_object
+#'
+#' @param ww_fit An object of class wwinference_fit
+#' @param ... additional arguments
+#'
+#' @family diagnostics
+#' @export
+parameter_diagnostics <- function(ww_fit, ...) {
```

Comment:
why do these nearly the same functions exist?

Reply:
Good point @gvegayon this is redundant from the function you created (`diagnostic_summary()`). I labeled this parameter diagnostics bc it helps me remember that this is the call that returns the diagnostics at the level of the individual parameter as opposed to the call to `cmdstan_obj$summary` that returns summaries across all parameters. Think we should just have one

Reply:
Sounds good, keeping parameter_diagnostics then.